src/java.desktop/share/classes/javax/swing/plaf/basic/BasicFileChooserUI.java

```diff
@@ -706,17 +709,19 @@ public void valueChanged(ListSelectionEvent evt) {
                         if (objects.length == 1
                             && ((File)objects[0]).isDirectory()
                             && chooser.isTraversable(((File)objects[0]))
-                            && (useSetDirectory || !fsv.isFileSystem(((File)objects[0])))) {
+                            && (useSetDirectory || (!fsv.isFileSystem(((File)objects[0]))
+                            && !Files.isSymbolicLink(Paths.get(((File)objects[0]).getPath()))))) {
```

Comment:
~~This does not align with other parts of the code where `isSymbolicLink` is combined together with `isFileSystem`.~~

You should format the code to avoid confusion: `isSymbolicLink` is not part of the top-level condition but rather of the part inside the parentheses after `||`

This should look this way:

```java
                            && (useSetDirectory
                                || (!fsv.isFileSystem(((File)objects[0]))
                                    && !Files.isSymbolicLink(((File)objects[0]).toPath())))) {
```